### PR TITLE
Fix sandslides and node destruction by falling nodes.

### PR DESCRIPTION
**Pull request by t4im**
This should fix both #3398 and #4594
(also confirmed to happen on a live-server of @LazyJ and @Miner48er for a while, last year or so?)

### CODE REVIEW

**Comment by paramat** [comment]
I wondered if it was due to ignore.

**Comment by paramat** [comment]
Because ignore is 'buildable to', ignore below a falling node allowed the falling node to fall into it.
The bug i experienced tended to happen when i was flying around, so un unloaded volume of sand may have been the cause.
Even if this doesn't fix my issue it is proper treatment of ignore.
Will test.

File: builtin/game/falling.lua [hunk 1]
```diff
@@ -49,11 +49,11 @@ core.register_entity(":__builtin:falling_node", {
...
 		local pos = self.object:getpos()
 		local bcp = {x = pos.x, y = pos.y - 0.7, z = pos.z} -- Position of bottom center point
-		local bcn = core.get_node(bcp)
-		local bcd = core.registered_nodes[bcn.name]
+		local bcn = core.get_node_or_nil(bcp)
+		local bcd = bcn and core.registered_nodes[bcn.name]
 		-- Note: walkable is in the node definition, not in item groups
-		if not bcd or
-				(bcd.walkable or
+		if bcn and
```
Comment (tenplus1) [inline builtin/game/falling.lua]: wouldn't this cancel out the following NOT bcd check?

**Comment by paramat** [comment]
Rebase needed.

**Comment by paramat** [comment]
On testing i had the same bug but this time underground with cave sand at y = -32.
I didn't experience the bug at the surface, but sometimes it's hard to reproduce.
However this is still correct treatment of ignore so :+1:

**Comment by paramat** [comment]
I think this helps, a surface sand collapse didn't happen over 20 mins, although sometimes it can be difficult to reproduce.
I still had those messages appear once but they may just be due to a large volume of sand collapsing, i visited the mapblock and found some underground cave sand but nothing strange.
So +1 on rebase.

**Comment by t4im** [comment]
> I still had those messages appear

There is, from what i can tell, still an additional bug, that is neither being addressed by this one, nor caused.
Falling nodes solidify, despite already having dropped, thereby duplicating it and causing these warnings.

When letting nodes fall over large distances, you should still find dropped items inside them.

**Comment by paramat** [comment]
I guess node objects falling on top of ignore will not revert to nodes so a large collapse onto ignore can result in force-delete if the player then moves away from the area.

**Comment by paramat** [comment]
@t4im needs rebase.

**Comment by paramat** [comment]
Might implement this myself as t4im is not around.

**Comment by paramat** [comment]
Replaced by #4740

**Pull request closed without merging**